drivers/adc/adc_mcux_lpadc.c

```diff
@@ -0,0 +1,401 @@
+	&& FSL_FEATURE_LPADC_HAS_CTRL_CALOFS
+#if defined(FSL_FEATURE_LPADC_HAS_OFSTRIM) \
+	&& FSL_FEATURE_LPADC_HAS_OFSTRIM
+	/* Request offset calibration. */
+#if defined(CONFIG_LPADC_DO_OFFSET_CALIBRATION) \
+	&& CONFIG_LPADC_DO_OFFSET_CALIBRATION
+	LPADC_DoOffsetCalibration(base);
+#else
+	LPADC_SetOffsetValue(base,
+			config->offset_a,
+			config->offset_b);
+#endif  /* DEMO_LPADC_DO_OFFSET_CALIBRATION */
+#endif  /* FSL_FEATURE_LPADC_HAS_OFSTRIM */
+	/* Request gain calibration. */
+	LPADC_DoAutoCalibration(base);
+#endif /* FSL_FEATURE_LPADC_HAS_CTRL_CALOFS */
+
+#if (defined(FSL_FEATURE_LPADC_HAS_CFG_CALOFS) \
+	&& FSL_FEATURE_LPADC_HAS_CFG_CALOFS)
+	/* Do auto calibration. */
+	LPADC_DoAutoCalibration(base);
+#endif /* FSL_FEATURE_LPADC_HAS_CFG_CALOFS */
+
+/* Enable the watermark interrupt. */
+#if (defined(FSL_FEATURE_LPADC_FIFO_COUNT) \
+	&& (FSL_FEATURE_LPADC_FIFO_COUNT == 2U))
+	LPADC_EnableInterrupts(base, kLPADC_FIFO0WatermarkInterruptEnable);
+#else
+	LPADC_EnableInterrupts(base, kLPADC_FIFOWatermarkInterruptEnable);
+#endif /* FSL_FEATURE_LPADC_FIFO_COUNT */
+
+	config->irq_config_func(dev);
+	data->dev = dev;
+
+	adc_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct adc_driver_api mcux_lpadc_driver_api = {
+	.channel_setup = mcux_lpadc_channel_setup,
+	.read = mcux_lpadc_read,
+#ifdef CONFIG_ADC_ASYNC
+	.read_async = mcux_lpadc_read_async,
+#endif
+};
+
+
+#define ASSERT_LPADC_CLK_SOURCE_VALID(val, str)	\
+	BUILD_ASSERT(val == 0 || val == 1 || val == 2 || val == 7, str)
+
+#define ASSERT_LPADC_CLK_DIV_VALID(val, str) \
+	BUILD_ASSERT(val == 1 || val == 2 || val == 4 || val == 8, str)
+
+#define ASSERT_LPADC_CALIBRATION_AVERAGE_VALID(val, str) \
+	BUILD_ASSERT(val == 1 || val == 2 || val == 4 || val == 8 \
+		|| val == 16 || val == 32 || val == 64 || val == 128, str) \
+
+#define ASSERT_WITHIN_RANGE(val, min, max, str)	\
+	BUILD_ASSERT(val >= min && val <= max, str)
+
+#define TO_LPADC_CLOCK_SOURCE(val) \
+	MUX_A(CM_ADCASYNCCLKSEL, val)
+#define TO_LPADC_REFERENCE_VOLTAGE(val) \
+	_DO_CONCAT(kLPADC_ReferenceVoltageAlt, val)
+#define TO_LPADC_CALIBRATION_AVERAGE(val) \
+	_DO_CONCAT(kLPADC_ConversionAverage, val)
+#define TO_LPADC_POWER_LEVEL(val) \
+	_DO_CONCAT(kLPADC_PowerLevelAlt, val)
+
+#define LPADC_MCUX_INIT(n)						\
+	static void mcux_lpadc_config_func_##n(const struct device *dev);	\
+									\
+	ASSERT_LPADC_CLK_SOURCE_VALID(DT_INST_PROP(n, clk_source),	\
+			  "Invalid clock source");			\
+	ASSERT_LPADC_CLK_DIV_VALID(DT_INST_PROP(n, clk_divider),	\
+		   "Invalid clock divider");			\
+	ASSERT_WITHIN_RANGE(DT_INST_PROP(n, voltage_ref), 2, 3,	\
+		"Invalid voltage reference source");	\
+	ASSERT_LPADC_CALIBRATION_AVERAGE_VALID(		\
+		DT_INST_PROP(n, calibration_average),	\
+		"Invalid converion average number for auto-calibration time");	\
+	ASSERT_WITHIN_RANGE(DT_INST_PROP(n, power_level), 1, 4,		\
+		"Invalid voltage reference source");					\
```

Comment:
It looks like this message was not updated after copy-pasting.

Reply:
Oops, I hadn't spotted that. I will correct it.